mslib/mswms/mss_plot_driver.py

```diff
@@ -676,3 +676,212 @@ def plot(self):
+                      left_longitude, self.lons.min())
+
+        # Shift the longitude field such that the data is in the range
+        # left_longitude .. left_longitude+360.
+        # NOTE: This does not overwrite self.lon_data (which is required
+        # in its original form in case other data is loaded while this
+        # file is open).
+        lon_data = ((self.lon_data - left_longitude) % 360) + left_longitude
+        lon_indices = lon_data.argsort()
+        lon_data = lon_data[lon_indices]
+        factors = []
+
+        # Make sure air_pressure is the first to be evaluated
+        variables = list(self.data_vars)
+        if variables[0] != "air_pressure":
+            variables.insert(0, variables.pop(variables.index("air_pressure")))
+
+        for name in variables:
+            var = self.data_vars[name]
+            data[name] = []
+            if len(var.shape) == 4:
+                var_data = var[timestep, ::-self.vert_order, ::self.lat_order, :]
+            else:
+                var_data = var[:][timestep, np.newaxis, ::self.lat_order, :]
+            logging.debug("\tLoaded %.2f Mbytes from data field <%s> at timestep %s.",
+                          var_data.nbytes / 1048576., name, timestep)
+            logging.debug("\tVertical dimension direction is %s.",
+                          "up" if self.vert_order == 1 else "down")
+            logging.debug("\tInterpolating to cross-section path.")
+            # Re-arange longitude dimension in the data field.
+            var_data = var_data[:, :, lon_indices]
+
+            cross_section = utils.interpolate_vertsec(var_data, self.lat_data, lon_data, self.lats, self.lons)
+            # Create vertical interpolation factors and indices for subsequent variables
+            if name == "air_pressure":
+                for index, alt in enumerate(self.alts):
+                    pressures = cross_section[:, index]
+                    closest = 0
+                    direction = 1
+                    for i, pressure in enumerate(pressures):
+                        if abs(pressure - alt) < abs(pressures[closest] - alt):
+                            closest = i
+                            direction = 1 if pressure - alt > 0 else -1
+                    next_closest = closest + direction if closest < len(pressures) - 1 else closest
+                    distance = abs(pressures[closest] - alt) + abs(pressures[next_closest] - alt)
+                    factors.append(
+                        [[closest, 1 - (abs(pressures[closest] - alt) / distance)],
+                         [next_closest, 1 - (abs(pressures[next_closest] - alt) / distance)]])
+                continue
+
+            for index in range(len(self.alts)):
+                cur_factor = factors[index]
+                value = cross_section[cur_factor[0][0], index] * cur_factor[0][1] + \
+                    cross_section[cur_factor[1][0], index] * cur_factor[1][1]
+                data[name].append(value)
```

Comment:
the whole thing looks like a performance nightmare for highly resolved data. The computation of factors is a O(n^2) operation in pure Python...
The pre-computation of factors *seems* like a good idea, but I ask myself if it would not be much faster to skip the precomputation do a repeated computation in C...

This should be marked as a TODO or issue once merged.

Potential improvements (once seen as necessary!):
*) use scipy.sparse to reformulate the second interpolation python-loop as a (sparse) matrix-vector product.
*) Use of https://unidata.github.io/MetPy/latest/api/generated/metpy.interpolate.interpolate_to_isosurface.html to handle the (full) interpolation. The (missing) third dimensions can be np.newaxis'd.
*) Either way, air_pressure does not need to be interpolated. One should move the computation of factors in front of the loop and simply use the target pressure values

Reply:
Marked as a TODO for now